Source/controls/devices/joystick.h

```diff
@@ -25,7 +26,10 @@ class Joystick {
 	static const std::vector<Joystick> &All();
 	static bool IsPressedOnAnyJoystick(ControllerButton button);
 
-	static ControllerButton ToControllerButton(const SDL_Event &event);
+	// Must be called exactly once at the start of each SDL input event.
+	void UnlockHatState();
+
+	static std::vector<ControllerButtonEvent> ToControllerButtonEvents(const SDL_Event &event);

```

Comment:
Since this can return at most 4 events, let's use `StaticVector<ControllerButtonEvent, 4>` to avoid allocations.